wpilibj/src/main/java/edu/wpi/first/wpilibj/PowerDistributionPanel.java

```diff
@@ -41,25 +50,63 @@ public PowerDistributionPanel() {
    * Query the input voltage of the PDP.
    *
    * @return The voltage of the PDP in volts
+   * @deprecated Will be replaced with a method that returns a Quantity of ElectricPotential in 2019
    */
+  @Deprecated
   public double getVoltage() {
     return PDPJNI.getPDPVoltage(m_module);
   }
 
+  /*
+  public Quantity<ElectricPotential> getVoltage() {
+    return Quantities.getQuantity(PDPJNI.getPDPVoltage(m_module), SI.VOLT);
+  }
+  */
+
+  /**
+   * Query the input voltage of the PDP.
+   *
+   * @param unit The unit to return as
+   * @return The electric potential of the PDP
+   */
+  public double getVoltage(Unit<ElectricPotential> unit) {
+    return Quantities.getQuantity(getVoltage(), SI.VOLT).to(unit).getValue().doubleValue();
+  }
+
   /**
    * Query the temperature of the PDP.
    *
    * @return The temperature of the PDP in degrees Celsius
+   * @deprecated Will be replaced with a method that returns a Quantity of Temperature in 2019
    */
+  @Deprecated
   public double getTemperature() {
     return PDPJNI.getPDPTemperature(m_module);
   }
 
+  /*
+  public Quantity<Temperature> getTemperature() {
+    return Quantities.getQuantity(PDPJNI.getPDPTemperature(m_module), SI.CELSIUS);
+  }
+  */
+
+  /**
+   * Query the temperature of the PDP.
+   *
+   * @param unit The unit to return as
+   * @return The temperature of the PDP
+   */
+  public double getTemperature(Unit<Temperature> unit) {
+    return Quantities.getQuantity(getTemperature(), SI.CELSIUS).to(unit).getValue().doubleValue();
```

Comment:
Why not Fahrenheit? Much more common for FRC teams. Or allow it to be customizable?

Reply:
Celsius is the temperature given by the HAL.